core/scripts/cre/environment/environment/environment.go

```diff
@@ -308,27 +306,17 @@ func startCmd() *cobra.Command {
 				return errors.Wrap(err, "failed to validate test configuration")
 			}
 
-			homeChainIDInt, chainErr := strconv.Atoi(in.Blockchains[0].ChainID)
-			if chainErr != nil {
-				return fmt.Errorf("failed to convert chain ID to int: %w", chainErr)
-			}
-
-			defaultCapabilities, defaultCapabilitiesErr := sets.NewDefaultSet(libc.MustSafeUint64FromInt(homeChainIDInt))
-			if defaultCapabilitiesErr != nil {
-				return errors.Wrap(defaultCapabilitiesErr, "failed to create default capabilities")
-			}
-
 			if err := validateWorkflowTriggerAndCapabilities(in, withExampleFlag, exampleWorkflowTrigger, withPluginsDockerImage); err != nil {
 				return errors.Wrap(err, "either cron binary path must be set in TOML config (%s) or you must use Docker image with all capabilities included and passed via withPluginsDockerImageFlag")
 			}
 
-			extraJobSpecFunctions := []cre.JobSpecFn{
-				// temporary solution until we figure out where that jobspec should live. Gateway is not a capability, it's more of a role
-				// but we don't have a good expression of that abstraction yet
-				gateway.JobSpec(append(extraAllowedGatewayPorts, in.Fake.Port), []string{}, []string{"0.0.0.0/0"}),
+			features := feature_set.New()
+			gatewayWhitelistConfig := gateway.WhitelistConfig{
```

Comment:
Can we move this as close as possible to where it's needed? i.e. whatever uses this should initialize it IMO;

What makes this difficult to do? (I see we rely on in.Fake.Port, is that it?)